src/pages/Integrations/SplunkSetup/SplunkSetupPage.tsx

```diff
@@ -0,0 +1,396 @@
+}
+
+2) Add user to group under /api/rbac/v1/groups/{uuid}/principals/ with the payload:
+
+{
+  "principals": [
+    {
+      "username": "smithj"
+    }
+  ]
+}
+
+3) Add role to group under /api/rbac/v1/groups/{uuid}/roles/ with the payload:
+
+{
+  "roles": [
+    "94846f2f-cced-474f-b7f3-47e2ec51dd11"
+  ]
+}
+
+4) [POST] Create Integrations under /api/integrations/v1.0/endpoints with the payload:
+{
+    "name": "Splunk Automation",
+    "enabled": true,
+    "type": "camel",
+    "sub_type": "splunk",
+    "description": "",
+    "properties": {
+        "url": "http://decd-187-3-186-244.ngrok.io",
+        "disable_ssl_verification": false,
+        "secret_token": "MYHEC_TOKEN",
+        "basic_authentication": {},
+        "extras": {}
+    }
+}
+
+5) Create behavior group under /api/notifications/v1.0/notifications/behaviorGroups with the payload:
+
+{
+  "bundle_id":"35fd787b-a345-4fe8-a135-7773de15905e",
+  "display_name":"Splunk-automation"
+}
+
+6) [POST] Update behavior group under api/notifications/v1.0/notifications/behaviorGroups/{BEHAVIOR_GROUP_ID}/actions with the payload:
+
+  ["8d8dca57-1834-48dd-b6ac-265c949c5e60"] <<-- Id of the integration
+
+7) [PUT] Update eventType under /api/notifications/v1.0/notifications/eventTypes/{EVENT_TYPE_UUID}/behaviorGroups with the payload:
+
+["ff59b502-da25-4297-bd88-6934ad0e0d63"] <<- Behavior group ID
+*/
+const RBAC_GROUPS_API = '/api/rbac/v1/groups/';
+const RBAC_ROLES_API = '/api/rbac/v1/groups/{}/roles/';
+const RBAC_ADD_USER_API = '/api/rbac/v1/groups/{}/principals/';
+const CREATE_INTEGRATION_API = '/api/integrations/v1.0/endpoints';
+const CREATE_BEHAVIOR_GROUP_API = '/api/notifications/v1.0/notifications/behaviorGroups';
+const UPDATE_BEHAVIOR_GROUP_API = '/api/notifications/v1.0/notifications/behaviorGroups/{}/actions';
+const EVENTTYPE_API = '/api/notifications/v1.0/notifications/eventTypes/{}/behaviorGroups';
+
+const NOTIF_ADM_ROLE_UUID = '15f910ad-150a-46dd-b666-0b21088e9c55'; // Stage UUID
+const SPLUNK_GROUP_NAME = 'SPLUNK_INTEGRATION';
+const SPLUNK_INTEGRATION_NAME = 'SPLUNK AUTOMATION';
+const SPLUNK_BEHAVIOR_GROUP_NAME = 'SPLUNK AUTOMATION GROUP';
+const BEHAVIOR_GROUP_BUNDLE_ID = '35fd787b-a345-4fe8-a135-7773de15905e';
+const EVENT_TYPES_IDS_ARR = ['6d4d1e03-f1bf-427c-8c62-08a114242b5d', // policy-triggered
+    '95a6d179-5dbd-4de4-a726-5bab2a699912',  // new-recommendation
+    '5df33474-9cfd-40e9-9a86-9f8857147f89', // drift-baseline-detected
+    '75233a10-646c-4e26-a5f1-639a7df8f29f']; // resolved-recommendation
+
+interface SplunkPageProps {
+    reduxDispatch: Dispatch;
+}
+
+export const SplunkSetupPage: React.FunctionComponent<SplunkPageProps> = props => {
+
+    const [ hecId, setHecId ] = useState('');
+    const [ splunkServerHostName, setHostName ] = useState('');
+    const [ automationLogs, setAutomationLogs ] = useState(`CLICK THE BUTTON TO START THE AUTOMATION\n`);
+
+    function handleHostInputChange(value) {
```

Comment:
Could you use arrow functions? Only asking for consistency across the code.
Also, some of these functions can be simplified:
i.e. 
```
const handleHostInputChange = setHostName; // or setHostName can be used directly
```